src/main/java/bio/terra/app/configuration/ApplicationConfiguration.java

```diff
@@ -444,4 +448,29 @@ public WebServerFactoryCustomizer<TomcatServletWebServerFactory> tomcatCustomize
             connector ->
                 connector.setEncodedSolidusHandling(EncodedSolidusHandling.DECODE.getValue()));
   }
+
+  @Bean("azureIPs")
+  public Map<Integer, List<String>> azureIPs() throws IOException {
+    URL url =
+        new URL(
+            "https://download.microsoft.com/download/7/1/D/71D86715-5596-4529-9B13-DA13A5DE5B63/ServiceTags_Public_20240708.json");
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode data = objectMapper.readTree(url);
+    JsonNode values = data.get("values");
+    // Map {Region ID: List of IP addresses}
```

Comment:
I'm mapping based on the region id here, since we may need to only allow IPs from certain regions for AXIN. I'm still trying to find the regions those ids correspond to...

Reply:
It looks like you can maybe use this endpoint to list the regions? https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations?view=rest-resources-2022-12-01&tabs=HTTP